scio-core/src/main/scala/com/spotify/scio/estimators/ApproxDistinctCounter.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.estimators
+
+import com.spotify.scio.coders.{Coder, CoderMaterializer}
+import com.spotify.scio.util.TupleFunctions._
+import com.spotify.scio.values.SCollection
+import org.apache.beam.sdk.{transforms => beam}
+
+/**
+ * Approximate distinct element counter for type `T`, e.g. HyperLogLog or HyperLogLog++. This has two APIs one
+ * estimate total distinct count for a given SCollection and second one estimate distinct count per each key in a
+ * key-value SCollection.
+ *
+ * @tparam T
+ */
+trait ApproxDistinctCounter[T] {
+
+  /**
+   * Return a SCollection with single (Long)value which is the estimated distinct count in the given SCollection with
+   * type `T`
+   */
+  def estimateDistinctCount(in: SCollection[T]): SCollection[Long]
+
+  /**
+   * Approximate distinct element per each key in the given key value SCollection.
+   * This will output estimated distinct count per each unique key.
+   */
+  def estimateDistinctCountPerKey[K](
+    in: SCollection[(K, T)]
+  )(implicit koder: Coder[K], voder: Coder[T]): SCollection[(K, Long)]
+}
+
+/**
+ * ApproxDistinctCounter impl for [[org.apache.beam.sdk.transforms.ApproximateUnique]] with sample size.
+ *
+ * Count approximate number of distinct values for each key in the SCollection.
+ * @param sampleSize the number of entries in the statistical sample; the higher this number, the
+ * more accurate the estimate will be; should be `>= 16`.
+ */
+case class ApproximateUniqueCounter[T](sampleSize: Int) extends ApproxDistinctCounter[T] {
```

Comment:
These 2 impls are the same as `SCollection#countApproxDistinct` & `PairSCollectionFunctions#countApproxDistinctByKey`?

EDIT: ah I see you deprecated those in favor of these "pluggable" ones. We don't really need to deprecate those though? They still function perfectly fine and has idiomatic names similar to Spark, etc. The new ones with `estimator: ApproxDistinctCounter[V]` can be optional for those who want to use HLL++.

Reply:
I didn't see much value of keeping those APIs, but if those APIs align with other frameworks, don't' see a reason to not support it. I will remove the deprecation note.

Reply:
Well it's best to not remove things unless necessary. These are supplemented by the 2 new methods, not replaced nor do they cause and confusion with the new ones.